Source/WebKit/NetworkProcess/cache/NetworkCacheDataCurl.cpp

```diff
@@ -113,11 +113,14 @@ RefPtr<SharedMemory> Data::tryCreateSharedMemory() const
     if (isNull() || !isMap())
         return nullptr;
 
-    auto newHandle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping();
-    if (!newHandle)
+    auto fileHandle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping();
+    if (!fileHandle)
         return nullptr;
 
-    return SharedMemory::adopt(newHandle.leak(), m_size, SharedMemory::Protection::ReadOnly);
+    auto dupHandle = fileHandle.copy();
```

Comment:
This code copies the handle twice. For `fileHandle` and `dupHandle`. Don't copy twice. And, check the handle after copying.